### ISSUE TITLE
fd-nested-list nested link a11y attribute can't be set using components

### ISSUE THREAD

**Issue by schaefermichael** [SAP/fundamental-ngx#8340]
#### Is this a bug, enhancement, or feature request?
Bug

#### Briefly describe your proposal.
We use components within our side navigation to automatically derive navigation items loaded by the router. With 0.35 the fd-nested-list adds a11y classes to nested lists and list items. However, that does not work if we generate the item link within a component. Getting rid of the custom component is currently not feasible for us, unfortunately.
We tried the following approaches:
- `<ul fd-nested-list><custom-component *ngFor>` with custom-component generating `<li fd-nested-list-item><a fd-nested-list-link>` (IMO bad because we have a DOM element between ul and li)
- `<ul fd-nested-list><li fd-nested-list-item custom-component *ngFor>` with custom-component generating `<a fd-nested-list-link>`
- `<ul fd-nested-list><li fd-nested-list-item *ngFor><custom-component>` with custom-component generating `<a fd-nested-list-link>`

All of those approaches generated an error here: https://github.com/SAP/fundamental-ngx/blob/main/libs/core/src/lib/nested-list/nested-list/nested-list.directive.ts#L166

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
Angular 14, fd-ngx/core 0.35.1

#### Please provide relevant source code if applicable.
https://stackblitz.com/edit/angular-ivy-kc3bg1

#### Is there anything else we should know?
A simple workaround would be to just skip the a11y classes if the link is not present. For us this is currently blocking our update process despite really wanting the updates as it solves a bug with the tokenizer.
